python/hail/dataset.py

```diff
@@ -5482,5 +5482,52 @@ def to_hail2(self):
         import hail2
         return hail2.VariantDataset(self.hc, self._jvds)
 
+    @handle_py4j
+    @record_method
+    @typecheck_method(pedigree=Pedigree,
+               complete_trios=bool)
+    def trio_matrix(self, pedigree, complete_trios=False):
```

Comment:
Hmm, this is a good point. I'll fix up pedigree and PR that to you separately, then rebase this on that.

Reply:
Is a pedigree conceptually more than just a list of trios?

Reply:
Right now it's just a list of trios, with some useful functions on lists of trios.

I can imagine it being useful to process a list of trios into a data structure that allows one to more effectively lookup the kind/degree of relatedness between samples given by the underlying "tree" (though if it's many generations there's bound to be inbreeding so not a tree). I don't think we should prioritize as nobody has asked for it, but I'd guess representing families trees comes up when searching for culprit in rare disease on a big case-control pedigree (which big is not actually big by study standards). For studies like UKBB I don't think this is all that relevant since the samples are from the same generation or two. But one might want to build/infer a pedigree from the data.

Reply:
I've poked around at it a bit, and I think it may be too difficult to change pedigree right now. Ideally it's just a list with some helper functions (or static functions on a list of trios), but we can't be manipulating lists of individual java objects right now for performance reasons. Are you strongly against leaving a flag for now?

Reply:
I am OK leaving this Pedigree change for another PR.